pkg/ingester/ingester_v2.go

```diff
@@ -1216,8 +1217,30 @@ func (i *Ingester) compactionLoop(ctx context.Context) error {
 	return nil
 }
 
+// compactHead compact the Head block at 2h intervals,
+// avoiding a single huge block.
+func compactHead(userDB *userTSDB) error {
+	// The Compact() will produces multiple 2h blocks if Head has a lot of data.
+	if err := userDB.Compact(); err != nil {
+		return err
+	}
+	twentyFourHrs := 24 * time.Hour.Milliseconds()
+	h := userDB.Head()
+	if (h.MinTime()/twentyFourHrs)*twentyFourHrs != (h.MaxTime()/twentyFourHrs)*twentyFourHrs {
```

Comment:
Shouldn't we check the configured `TSDBConfig.BlockRanges` and compact the head if mint and maxt don't belong to the same range period?

Reply:
That would be a better one